conda_libmamba_solver/solver.py

```diff
@@ -171,21 +172,23 @@ def solve_final_state(
         else:
             IndexHelper = LibMambaIndexHelper
 
-        if os.getenv("CONDA_LIBMAMBA_SOLVER_NO_CHANNELS_FROM_INSTALLED") or (
-            getattr(context, "_argparse_args", None) or {}
-        ).get("override_channels"):
-            # see https://github.com/conda/conda-libmamba-solver/issues/108
-            channels_from_installed = ()
-        else:
-            channels_from_installed = in_state.channels_from_installed()
-
-        all_channels = (
+        all_channels = [
             *conda_bld_channels,
             *self.channels,
             *in_state.channels_from_specs(),
-            *channels_from_installed,
-            *in_state.maybe_free_channel(),
-        )
+        ]
+        override = (getattr(context, "_argparse_args", None) or {}).get("override_channels")
+        if not os.getenv("CONDA_LIBMAMBA_SOLVER_NO_CHANNELS_FROM_INSTALLED") and not override:
+            # see https://github.com/conda/conda-libmamba-solver/issues/108
+            all_urls = [url for c in all_channels for url in Channel(c).urls(False)]
+            installed_channels = in_state.channels_from_installed(seen=all_urls)
+            for channel in installed_channels:
+                # Only add to list if resource is available; check has timeout=1s
+                if timeout(1, is_channel_available, channel.base_url, default_return=False):
```

Comment:
Note this `timeout` function is a no-op on Windows. If those channels happen to be network mounts or the like, this can be a bottleneck. Hopefully, those cases are very rare.

Reply:
Maybe the timeout check should be part of the function it self, with a kwarg?

Reply:
I suggest we add this to [`conda` itself](https://github.com/conda/conda/blob/dea94d4a339c35f531d686108862002054222153/conda/common/io.py#L349-L355).

Reply:
Filed https://github.com/conda/conda/issues/13131 and opened https://github.com/conda/conda/pull/13132